src/Microsoft.SqlTools.ServiceLayer/Metadata/SmoScripterFactory.cs

```diff
@@ -0,0 +1,191 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.SqlCore.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.ReliableConnection;
+
+
+namespace Microsoft.SqlTools.ServiceLayer.Metadata
+{
+    internal static class SmoScripterFactory
+    {
+        public static async Task<IEnumerable<string>> GenerateAllServerTableScripts(DbConnection connection)
+        {
+            var serverConnection = SmoScripterFactory.GetServerConnection(connection);
+            if (serverConnection == null)
+            {
+                return null;
```

Comment:
You don't want to return an empty enumerable instead? If you do return this it'd be nice to type this correctly (`Task<IEnumerable<string>?>`